### PR TITLE
Add support for deploying to Docker Hub

### LINKED ISSUE #18: Make Docker Container Open Source
As a developer in the OSCAL community, I want to be able to access and contribute to the OSCAL Editor so that I can view and edit OSCAL files without building and configuring separate components.

Note this is blocked by EasyDynamics/easygrc#25.

### Acceptance Criteria
- The `oscal-editor-deployment` GitHub repo is public
- A license has been specified
- Copyright has been defined if not public domain
- A contribution page exists
- The Docker container is available on Docker Hub (as a result of an automated pipeline)
  - Note that this should be the **last** part of implementation of this story, i.e. should not hold up making the repository open source and public.  If so, we can move this to a future story.

### Implementation Notes
- Contributing and code of conduct should pull from our organization default

### LINKED ISSUE #37: Add a GitHub Actions Workflow to Test Deployment PRs
As a developer contributing to or receiving contributions on the OSCAL Editor Deployment, I want to be able to test PRs through an automated workflow so that I can have confidence that the changes won't break functionality.

Now that the repository is public, we should have a workflow that builds the container and runs the end-to-end tests as part of the PR process. 

### Acceptance Criteria
 - A basic workflow is added to ensure that the Docker container builds and that the end-to-end tests run on every PR

### Implementation Notes
- This *may* depend on #17 so that we can pull from a branch of `oscal-rest-service` rather than relying on a token with `packages:read` (but we may not need that anymore since `oscal-rest-service` is public).

### PR DESCRIPTION
**Pull request by laurelmay**
This consolidates the workflows for PRs and other actions into a single workflow (since the steps are largely the same). For pushes to `main` (or tags), we will perform the deployment steps after the testing is complete. For PRs (such as this one), we stop after the tests.

Since the related repositories are now public, we can stop using the `PACKAGE_READ_TOKEN` and just leverage the default `GITHUB_TOKEN` for fetching packages.

### CODE REVIEW

File: .github/workflows/container-test-deploy.yml [hunk 1]
```diff
@@ -1,7 +1,11 @@
 name: Build, Test, Deploy
 on:
   push:
-    branches: [main]
+    branches:
+      - main
+    tags:
+      - 'v*'
```
Comment (rgauss) [inline .github/workflows/container-test-deploy.yml:7]: Looks like this is required by `type=semver` and the consequence is that we have to use `vX.X.X` as our tag syntax, which is fine I guess.

**Review by rgauss — APPROVED** [review]

**Comment by rgauss** [comment]
I believe this also closes #37.

File: .github/workflows/container-test-deploy.yml (hunk 1 above)
Comment (kylelaker) [reply .github/workflows/container-test-deploy.yml:7]: Yeah, we don't necessarily need to use `type=semver` and can always adjust that usage in the future if we choose and do more custom labeling. But I think `vX.X.X` is reasonable if we're doing semver. I think that's what we do on `oscal-react-library` and follows GitHub's recommendations on how to name tags.

From https://github.com/EasyDynamics/oscal-editor-deployment/releases/new:
image

If we don't want to semver then we can do something a little different/smarter here over time.

Comment (rgauss) [reply .github/workflows/container-test-deploy.yml:7]: No, it's fine as is.

**Review by zclarkEDC — APPROVED** [review]

**Pull request merged**